mlflow/keras.py

```diff
@@ -751,9 +751,20 @@ def fit(self, *args, **kwargs):
         return _run_and_log_function(self, original, args, kwargs, unlogged_params, 5)
 
     def fit_generator(self, *args, **kwargs):
+        """
+        NOTE: `fit_generator()` is deprecated in Keras >= 2.4.0 and simply wraps `fit()`.
+        To avoid unintentional creation of nested MLflow runs caused by a patched
+        `fit_generator()` method calling a patched `fit()` method, we only patch
+        `fit_generator()` in Keras < 2.4.0.
+        """
         original = gorilla.get_original_attribute(keras.Model, "fit_generator")
         unlogged_params = ["self", "generator", "callbacks", "validation_data", "verbose"]
         return _run_and_log_function(self, original, args, kwargs, unlogged_params, 4)
 
     wrap_patch(keras.Model, "fit", fit)
-    wrap_patch(keras.Model, "fit_generator", fit_generator)
+    # `fit_generator()` is deprecated in Keras >= 2.4.0 and simply wraps `fit()`.
```

Comment:
Interestingly, Keras >= 2.4.0 pulls in its `Model` class definition from `tensorflow.keras` with a dependency on `tensorflow >= 2.2.0`. This explains why `fit_generator()` is deprecated in Keras 2.4.0, since `tensorflow` deprecated it in TensorFlow 2.1.0.